core/src/main/scala/spinal/core/sim/SimBootstraps.scala

```diff
@@ -564,7 +564,14 @@ abstract class SimCompiled[T <: Component](val report: SpinalReport[T]){
 
   def newSimRaw(name: String, seed: Int) : SimRaw
 
-  def doSimApi(name: String = "test", seed: Int = sys.env.getOrElse("SPINAL_SIM_SEED_RANDOM", "0").toInt*Random.nextInt(2000000000), joinAll: Boolean)(body: T => Unit): Unit = {
+  def newSeed(): Int = {
+    sys.env.get("SPINAL_SIM_SEED") match {
+      case Some(v) => v.toInt
+      case None => Random.nextInt(2000000000)
```

Comment:
I am always curious on each const value, how this constant value is derived? And why it should be a const?
Although, it is from the legacy code here. :)

Reply:
same here, no idea why that number so I kept it ;-)

Reply:
I just didn't wanted a negative number, that's why i set it to something arbitrary "big".

Reply:
Ah then do this?

```scala
Random.nextInt(Integer.MAX_VALUE);
```

Reply:
Yes you are right ^^

Reply:
I pushed Integer.MAX_VALUE in the repo.